test/cpp_extensions/libtorch_agnostic_extension/libtorch_agnostic/csrc/kernel.cpp

```diff
@@ -476,14 +476,59 @@ void boxed_test_default_constructor(
   stack[0] = from(res);
 }
 
+uint64_t get_any_data_ptr(Tensor t, bool mutable_) {
+  if (mutable_) {
+    return reinterpret_cast<uint64_t>(t.mutable_data_ptr());
+  } else {
+    return reinterpret_cast<uint64_t>(t.const_data_ptr());
+  }
+}
+
+void boxed_get_any_data_ptr(
+    StableIValue* stack,
+    uint64_t num_args,
+    uint64_t num_outputs) {
+  uint64_t res = get_any_data_ptr(to<Tensor>(stack[0]), to<bool>(stack[1]));
+  stack[0] = from(res);
+}
+
+uint64_t get_template_any_data_ptr(Tensor t, Tensor r, bool mutable_) {
```

Comment:
nit: could the ScalarType be passed directly? why do we need to use a tensor to carry it

Reply:
yes, passing ScalarType works now (after moving ScalarType to headeronly). I'll update tests!